gaas/src/main/kotlin/tw/waterballsa/utopia/gaas/AbsenceRecord.kt

```diff
@@ -0,0 +1,135 @@
+package tw.waterballsa.utopia.gaas
+
+import net.dv8tion.jda.api.entities.ScheduledEvent
+import net.dv8tion.jda.api.events.guild.scheduledevent.ScheduledEventCreateEvent
+import net.dv8tion.jda.api.events.guild.scheduledevent.update.ScheduledEventUpdateStatusEvent
+import tw.waterballsa.utopia.commons.config.WsaDiscordProperties
+import tw.waterballsa.utopia.jda.listener
+import tw.waterballsa.utopia.jda.log
+import java.lang.System.lineSeparator
+import java.nio.file.Files.*
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption.APPEND
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.*
+import kotlin.math.roundToInt
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+private val studyCircleEventIds = mutableListOf<String>()
+private val timer = Timer()
+private const val TIMEZONE_ID = "Asia/Taipei"
+private const val DATABASE_PATH = "data/gaas"
+private const val DATAFILE_NAME_TEMPLATE = "/study-circle-absence-record-\$date.db"
+private val dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")
+
+
+fun collectStudyCircleEvent(wsaDiscordProperties: WsaDiscordProperties) = listener {
+    on<ScheduledEventCreateEvent> {
+        val partyChannelId = wsaDiscordProperties.wasPartyChannelId
+        scheduledEvent
+            .takeIf { it.name.contains("遊戲微服務") && it.channel?.id == partyChannelId }
+            ?.run {
+                studyCircleEventIds.add(id)
+                log.info { "Successfully collect the event of study circle, event id: $id" }
```

Comment:
please follow the logging format `[<event>] <data in json format>`
for example `[Message Sent] {\"content\": \"...\"}`